pkg/aws/client/client.go

```diff
@@ -364,6 +364,19 @@ func (c *Client) CreateBucketIfNotExists(ctx context.Context, bucket, region str
 		return err
 	}
 
+	// Block public access to the bucket
+	if _, err := c.S3.PutPublicAccessBlockWithContext(ctx, &s3.PutPublicAccessBlockInput{
+		Bucket: aws.String(bucket),
+		PublicAccessBlockConfiguration: &s3.PublicAccessBlockConfiguration{
+			BlockPublicAcls:       aws.Bool(true),
+			BlockPublicPolicy:     aws.Bool(true),
+			IgnorePublicAcls:      aws.Bool(true),
+			RestrictPublicBuckets: aws.Bool(true),
+		},
+	}); err != nil {
+		return err
+	}
+
```

Comment:
I think we should add bucket policy here. I received some alert for not having proper bucket policies. Example bucket policy:
```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "RestrictToTLSRequestsOnly",
            "Effect": "Deny",
            "Principal": "*",
            "Action": "s3:*",
            "Resource": "arn:aws:s3:::9eeccd1a-d2d1-499b-977c-0d346f9d3b79/*",
            "Condition": {
                "Bool": {
                    "aws:SecureTransport": "false"
                },
                "NumericLessThan": {
                    "s3:TlsVersion": "1.2"
                }
            }
        }
    ]
}
```
More info: https://docs.aws.amazon.com/AmazonS3/latest/userguide/access-policy-language-overview.html